packages/starlight/integrations/shared/pathToLocale.ts

```diff
@@ -22,11 +22,11 @@ export function pathToLocale(
 ): string | undefined {
 	const srcDir = new URL(astroConfig.srcDir, astroConfig.root);
 	const docsDir = new URL('content/docs/', srcDir);
-	const slug = path
-		// Format path to unix style path.
-		?.replace(/\\/g, '/')
-		// Strip docs path leaving only content collection file ID.
-		// Example: /Users/houston/repo/src/content/docs/en/guide.md => en/guide.md
-		.replace(docsDir.pathname, '');
+	// Format path to unix style path.
+	path = path?.replace(/\\/g, '/')
+	if (path && !path.startsWith('/') && docsDir.pathname.startsWith('/')) path = '/' + path;
```

Comment:
Can we add a comment here about why this step is necessary. You explained it to me, but I’ve already forgotten, so seems like a good sign a comment is a good idea 😁

Reply:
Sure, will do! It's a Windows-specific issue where the input path does not have a leading slash, and therefore your "remove the matching beginning" logic doesn't work otherwise.

Reply:
Done! :)